EIPS/eip-191.md

```diff
@@ -55,29 +54,54 @@ Using `0x19` thus makes it possible to extend the scheme by defining a version `
 |    `0x01`    | [712][eip-712] | Structured data
 |    `0x45`    | [191][eip-191] | `personal_sign` messages
 
+#### Version `0x00`
+
+```
+0x19 <0x00> <intended validator address> <data to sign>
+```
+
+The version `0x00` has `<intended validator address>` for the version specific data. In the case of a Multisig wallet that perform an execution based on a passed signature, the validator address is the address of the Multisig itself. The data to sign could be any arbitrary data.
+
+#### Version `0x01`
+
+The version `0x01` is for structured data as defined in [EIP-712]
+
+#### Version `0x45` (E)
+
+```
+0x19 <0x45 (E)> <thereum Signed Message:\n" + len(message)> <data to sign>
+```
+
+The version `0x45` (E) has `<thereum Signed Message:\n" + len(message)>` for the version specific data. The data to sign could be any arbitrary data.
+
 [EIP-191]: ./eip-191.md
 [EIP-712]: ./eip-712.md
 
 ### Example
 
-The following snippet has been written in Solidity 0.5.0.
+The following snippets has been written in Solidity 0.8.0.
+
+#### Version `0x00`
 
 ```solidity
-function submitTransactionPreSigned(address destination, uint value, bytes data, uint nonce, uint8 v, bytes32 r, bytes32 s)
-    public
-    returns (bytes32 transactionHash)
-{
+function signatureBasedExecution(address target, uint256 nonce, bytes memory payload, bytes memory signature)
+    public payable {
+        
     // Arguments when calculating hash to validate
     // 1: byte(0x19) - the initial 0x19 byte
     // 2: byte(0) - the version byte
-    // 3: this - the validator address
-    // 4-7 : Application specific data
-    transactionHash = keccak256(abi.encodePacked(byte(0x19),byte(0),address(this),destination, value, data, nonce));
-    sender = ecrecover(transactionHash, v, r, s);
-    // ...
+    // 3: address(this) - the validator address
+    // 4-6 : Application specific data
+
+    bytes32 hash = keccak256(abi.encodePacked(byte(0x19), byte(0), address(this), msg.value, nonce, payload));
+
+    // recovering the signer from the hash and the signature
+    addressRecovered = ECDSA.recover(hash, signature);
```

Comment:
Hm, I'm leery about changing this from plain `ecrecover` to what I assume is supposed to be OZ's. Their implementation has additional restrictions that plain `ecrecover` doesn't have. Changes to final EIPs are supposed to be backwards compatible.